src/main/java/org/zalando/riptide/Rest.java

```diff
@@ -69,16 +73,23 @@ public Dispatcher execute(final HttpMethod method, final URI url, final HttpHead
      * response thrown by the {@link OAuth2CompatibilityResponseErrorHandler} and continues with normal dispatching.
      * </p>
      */
-    private <T> ClientHttpResponse execute(final HttpMethod method, final URI url, final Callback<T> callback) {
+    private <T> ClientHttpResponse executeRequest(final HttpMethod method, final URI url, HttpEntity<T> entity) {
         try {
-            return template.execute(url, method, callback, BufferingClientHttpResponse::buffer);
-        } catch (final AlreadyConsumedResponseException e) {
-            return e.getResponse();
+            final ClientHttpRequest request = clientHttpRequestFactory.createRequest(url, method);
+            RequestUtil.writeRequestEntity(entity, request, converters);
+            return request.execute();
+        } catch (IOException ex) {
+            final String message = String.format("I/O error on %s request for \"%s\": %s", method.name(), url, ex.getMessage());
+            throw new ResourceAccessException(message, ex);
         }
     }
 
+    public static Rest create(final ClientHttpRequestFactory clientHttpRequestFactory, final List<HttpMessageConverter<?>> converters) {
+        return new Rest(clientHttpRequestFactory, converters);
+    }
+
     public static Rest create(final RestTemplate template) {
-        return new Rest(template);
+        return create(template.getRequestFactory(), template.getMessageConverters());
```

Comment:
> Which the tokens library is doing for us anyway, right?

> True. But we could make a strong argument for it in our README.

Yes but with a slightly different semantic (constant refreshing vs. on-demand). Also [this discussion](https://github.com/zalando/riptide/pull/104/files#r66149691) relates to this. If we drop `RestTemplate` from the API users will not be confused about using `RestTemplate` vs. `OAuth2RestTemplate` vs. `StupsOAuth2RestTemplate` and would solve it own their own.